react/features/invite/components/dial-in-summary/NumbersList.web.js

```diff
@@ -41,76 +41,144 @@ class NumbersList extends Component<Props> {
+            </div>);
     }
 
     /**
      * Renders rows of countries and associated phone numbers.
      *
-     * @param {Object} numbersMapping - An object with country names as keys
-     * and values as arrays of phone numbers.
+     * @param {Object|Array<Object>} numbersMapping - An object with country
+     * names as keys and values as arrays of phone numbers.
      * @private
      * @returns {ReactElement[]}
      */
-    _renderWithCountries(numbersMapping: Object) {
-        const rows = [];
+    _renderWithCountries(
+            numbersMapping: { numbers: Array<string> } | Array<Object>) {
+        const { t } = this.props;
+        let numbers;
+
+        if (Array.isArray(numbersMapping)) {
+            numbers = numbersMapping.reduce(
+                (resultNumbers, number) => {
+                    const countryName
+                        = t(`countries:countries.${number.countryCode}`);
+
+                    if (resultNumbers[countryName]) {
+                        resultNumbers[countryName].push(number);
+                    } else {
+                        resultNumbers[countryName] = [ number ];
+                    }
+
+                    return resultNumbers;
+                }, {});
+        } else {
+            numbers = {};
+
+            for (const [ country, numbersArray ]
+                of Object.entries(numbersMapping.numbers)) {
+                if (!Array.isArray(numbersArray)) {
+                    return;
+                }
+
+                /* eslint-disable arrow-body-style */
+                const formattedNumbers = numbersArray.map(number => ({
+                    formattedNumber: number
+                }));
+                /* eslint-enable arrow-body-style */
 
-        for (const [ country, numbers ] of Object.entries(numbersMapping)) {
-            if (!Array.isArray(numbers)) {
-                return;
+                numbers[country] = formattedNumbers;
             }
+        }
 
-            const formattedNumbers = numbers.map(number => {
-                if (typeof number === 'string') {
-                    return this._renderNumberDiv(number);
-                }
+        let hrKey = 0;
+        const rows = [ <hr key = { hrKey++ } /> ];
 
-                return null;
-            });
+        Object.keys(numbers).forEach((countryName: string) => {
+            const numbersArray = numbers[countryName];
 
             rows.push(
-                <tr key = { country }>
-                    <td>{ country }</td>
-                    <td className = 'dial-in-numbers'>{ formattedNumbers }</td>
-                </tr>
+                <div
+                    className = 'number-group'
+                    key = { countryName }>
+                    {this._renderFlag(numbersArray[0].countryCode)}
+                    <div className = 'numbers-and-country' >
+                        <div className = 'country' > { countryName } </div>
```

Comment:
Did linting not complain about missing or extra spaces?

Reply:
Nope.